### ISSUE TITLE
Architecture Clarification

### ISSUE THREAD

**Issue by parhartanvir** [affinelayer/pix2pix-tensorflow#28]
It's more of a doubt than an issue. Can you tell he why does the graph have two discriminators, viz. real_discriminator and fake_discriminator. 
I could not know from the paper or your [blog post](https://affinelayer.com/pix2pix/).
Is this how you update wrights separately for images generated by generator and the real images?
In that case are the real/fake inputs to the discriminator given at random or in a particular order?

By the way, thanks for the implementation! 
Tanvir

**Comment by Cristy94** [comment]
Yes, they are trained separately (eg. there are two discriminators, one for real, one from fake).

From the source code:
```
    with tf.name_scope("discriminator_loss"):
        # minimizing -tf.log will try to get inputs to 1
        # predict_real => 1
        # predict_fake => 0
        discrim_loss = tf.reduce_mean(-(tf.log(predict_real + EPS) + tf.log(1 - predict_fake + EPS)))
```

You can see how the final discriminator loss is obtained by combining the output of the two discriminators. The loss is with `minus` for the `predict_real` discriminator (eg. if it said it is fake but it was actually real it's not good).

PS: I just started learning about DL and using this library for a couple days, I might be wrong.

**Comment by WLpub** [comment]
I just start to read this code, and I think the point is:

with tf.variable_scope("discriminator", reuse=True):

two discriminators use and train same weights.
